### ISSUE TITLE
modelimport module, when I import a keras model, no PreProcessor for InputTypeFeedForward(32)

### ISSUE THREAD

**Issue by liuyichaosoftware** [deeplearning4j/deeplearning4j#4521]
#### Issue Description

my keras model Input is an Embedding layer,
but when imported, I got an null Exception,

Exception in thread "main" java.lang.NullPointerException
	at org.deeplearning4j.nn.modelimport.keras.layers.core.KerasFlatten.getOutputType(KerasFlatten.java:71)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.inferOutputTypes(KerasModel.java:262)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:165)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:92)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildModel(KerasModelBuilder.java:115)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:179)

because no Preprocesor for it.

#### Version Information

master

#### Contributing

```
else if (inputType[0] instanceof InputTypeRecurrent) {
                preprocessor = new RnnToFeedForwardPreProcessor();
            } else if (inputType[0] instanceof InputTypeFeedForward){
                preprocessor = new  *****
            }
```
I think some thing need be added in KerasFlatten

**Comment by maxpumperla** [comment]
@liuyichaosoftware can you please calm down a little. :) Again, what's the model you're using. Please post json or h5 as gist.

**Comment by liuyichaosoftware** [comment]
ok, my json is as follow.

`{"class_name": "Model", "keras_version": "2.0.8", "config": {"layers": [{"class_name": "InputLayer", "config": {"dtype": "float32", "batch_input_shape": [null, 60], "name": "input_1", "sparse": false}, "inbound_nodes": [], "name": "input_1"}, {"class_name": "InputLayer", "config": {"dtype": "float32", "batch_input_shape": [null, 1], "name": "input_2", "sparse": false}, "inbound_nodes": [], "name": "input_2"}, {"class_name": "Embedding", "config": {"embeddings_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "embedding_1", "dtype": "float32", "output_dim": 128, "trainable": true, "embeddings_regularizer": {"class_name": "L1L2", "config": {"l2": 9.999999747378752e-06, "l1": 0.0}}, "input_dim": 10001, "mask_zero": true, "embeddings_constraint": null, "batch_input_shape": [null, null], "activity_regularizer": null, "input_length": null}, "inbound_nodes": [[["input_1", 0, 0, {}]]], "name": "embedding_1"}, {"class_name": "Embedding", "config": {"embeddings_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "embedding_2", "dtype": "float32", "output_dim": 32, "trainable": true, "embeddings_regularizer": {"class_name": "L1L2", "config": {"l2": 9.999999747378752e-06, "l1": 0.0}}, "input_dim": 24, "mask_zero": false, "embeddings_constraint": null, "batch_input_shape": [null, null], "activity_regularizer": null, "input_length": null}, "inbound_nodes": [[["input_2", 0, 0, {}]]], "name": "embedding_2"}, {"class_name": "LSTM", "config": {"recurrent_activation": "hard_sigmoid", "trainable": true, "recurrent_initializer": {"class_name": "Orthogonal", "config": {"seed": null, "gain": 1.0}}, "use_bias": true, "bias_regularizer": null, "return_state": false, "unroll": false, "activation": "tanh", "bias_initializer": {"class_name": "Zeros", "config": {}}, "units": 128, "unit_forget_bias": true, "activity_regularizer": null, "recurrent_dropout": 0.0, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "kernel_constraint": null, "dropout": 0.0, "stateful": false, "recurrent_regularizer": null, "name": "lstm_1", "bias_constraint": null, "go_backwards": false, "implementation": 0, "kernel_regularizer": null, "return_sequences": false, "recurrent_constraint": null}, "inbound_nodes": [[["embedding_1", 0, 0, {}]]], "name": "lstm_1"}, {"class_name": "Flatten", "config": {"trainable": true, "name": "flatten_1"}, "inbound_nodes": [[["embedding_2", 0, 0, {}]]], "name": "flatten_1"}, {"class_name": "Dense", "config": {"kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "dense_1", "kernel_constraint": null, "bias_regularizer": null, "bias_constraint": null, "activation": "relu", "trainable": true, "kernel_regularizer": {"class_name": "L1L2", "config": {"l2": 9.999999747378752e-06, "l1": 0.0}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "units": 64, "use_bias": true, "activity_regularizer": null}, "inbound_nodes": [[["lstm_1", 0, 0, {}]]], "name": "dense_1"}, {"class_name": "Dense", "config": {"kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "dense_2", "kernel_constraint": null, "bias_regularizer": null, "bias_constraint": null, "activation": "relu", "trainable": true, "kernel_regularizer": {"class_name": "L1L2", "config": {"l2": 9.999999747378752e-06, "l1": 0.0}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "units": 8, "use_bias": true, "activity_regularizer": null}, "inbound_nodes": [[["flatten_1", 0, 0, {}]]], "name": "dense_2"}, {"class_name": "Dropout", "config": {"rate": 0.1, "trainable": true, "name": "dropout_1"}, "inbound_nodes": [[["dense_1", 0, 0, {}]]], "name": "dropout_1"}, {"class_name": "Dropout", "config": {"rate": 0.1, "trainable": true, "name": "dropout_2"}, "inbound_nodes": [[["dense_2", 0, 0, {}]]], "name": "dropout_2"}, {"class_name": "Concatenate", "config": {"trainable": true, "name": "concatenate_1", "axis": -1}, "inbound_nodes": [[["dropout_1", 0, 0, {}], ["dropout_2", 0, 0, {}]]], "name": "concatenate_1"}, {"class_name": "Dense", "config": {"kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "dense_3", "kernel_constraint": null, "bias_regularizer": null, "bias_constraint": null, "activation": "softmax", "trainable": true, "kernel_regularizer": {"class_name": "L1L2", "config": {"l2": 9.999999747378752e-06, "l1": 0.0}}, "bias_initializer": {"class_name": "Zeros", "config": {}}, "units": 10001, "use_bias": true, "activity_regularizer": null}, "inbound_nodes": [[["concatenate_1", 0, 0, {}]]], "name": "dense_3"}], "input_layers": [["input_1", 0, 0], ["input_2", 0, 0]], "output_layers": [["dense_3", 0, 0]], "name": "model_1"}, "backend": "tensorflow"}
`
thank you

**Comment by maxpumperla** [comment]
@liuyichaosoftware ok, so I figured out that this is related to #4433. In DL4J, Embedding layers produce 2D output, hence are of feed-forward type. in this case flatten really doesn't have to do anything, so that is correct. however, we have to fix #4433, after which your model should work.

I'll close this as a duplicate for now, but please feel free to re-open if it still doesn't work after fixing the above issue.

**Comment by liuyichaosoftware** [comment]
OK， thank you～

**Comment by liuyichaosoftware** [comment]
OK， thank you~
